### ISSUE TITLE
GO issue 'cannot find package "github.com/bitrise-io/go-utils/cmdex' while running old version of GO and CLI 1.4.2

### ISSUE THREAD

**Issue by pupca** [bitrise-io/bitrise#426]
Hi! 
_scroll down for TL;DR_

when running `bitrise --ci=true run Beta`

with bitrise.yaml:

```
---
format_version: 1.3.0
default_step_lib_source: https://github.com/bitrise-io/bitrise-steplib.git
trigger_map:
- pull_request_source_branch: beta_build
  workflow: Beta
  pull_request_target_branch: "*"
- push_branch: beta_build
  workflow: Beta
- pull_request_source_branch: app_build
  pull_request_target_branch: "*"
  workflow: App
- push_branch: app_build
  workflow: App
workflows:
  Beta:
    steps:
    - activate-ssh-key@3.1.1:
        run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
    - git-clone@3.4.0:
        inputs:
        - clone_depth: '1'
    - set-xcode-build-number@1.0.4:
        inputs:
        - plist_path: source/osx/Resources/Info.plist
        - build_version_offset: ''
    - certificate-and-profile-installer@1.7.0: {}
    - xcode-archive-mac@1.3.1:
        inputs:
        - configuration: Beta
        - export_method: developer-id
    - deploy-to-bitrise-io@1.2.5: {}
    - hockeyapp-deploy@2.3.0:
        inputs:
        - ipa_path: "$BITRISE_EXPORTED_FILE_PATH"
        - api_token: XXX
        - app_id: XXX
        - notes: "$GIT_CLONE_COMMIT_MESSAGE_BODY"
    - script@1.1.3:
        inputs:
        - content: |-
            #!/bin/bash
            # fail if any commands fails
            set -e
            # debug log
            set -x

            mkdir -p /Users/vagrant/beta
            cp /Users/vagrant/deploy/Beta.app.zip /Users/vagrant/beta/XXX.zip
    - amazon-s3-uploader@1.0.1:
        inputs:
        - file_path: "/Users/vagrant/beta/XXX.zip"
        - aws_access_key: XXX
        - aws_secret_key: XXX
        - bucket_name: XXX
        - path_in_bucket: beta
    - slack@2.3.0:
        inputs:
        - webhook_url: XXX
        - channel: "#builds"
        - from_username: XX BETA Build - Success
        - from_username_on_error: XX BETA Build - ERROR
        - message: |-
            Yay! New build is done! Here are results:

            Direct Download: <XXX>

            HockeyApp: <$HOCKEYAPP_DEPLOY_PUBLIC_URL>
        - message_on_error: |-
            Build Failed :(

            Bitrise.io: <$BITRISE_BUILD_URL>
    before_run: 
    after_run: 
  App:
    steps:
    - activate-ssh-key@3.1.1:
        run_if: '{{getenv "SSH_RSA_PRIVATE_KEY" | ne ""}}'
    - git-clone@3.4.0:
        inputs:
        - clone_depth: '1'
    - set-xcode-build-number@1.0.4:
        inputs:
        - plist_path: source/osx/Resources/Info.plist
        - build_version_offset: ''
    - certificate-and-profile-installer@1.7.0: {}
    - xcode-archive-mac@1.3.1:
        inputs:
        - scheme: Production
        - configuration: Production
        - export_method: developer-id
    - deploy-to-bitrise-io@1.2.5: {}
    - hockeyapp-deploy@2.3.0:
        inputs:
        - ipa_path: "$BITRISE_EXPORTED_FILE_PATH"
        - api_token: XXX
        - app_id: XXX
        - notes: "$GIT_CLONE_COMMIT_MESSAGE_BODY"
    - script@1.1.3:
        inputs:
        - content: |-
            #!/bin/bash
            # fail if any commands fails
            set -e
            # debug log
            set -x

            mkdir -p /Users/vagrant/app
            cp /Users/vagrant/deploy/Production.app.zip /Users/vagrant/app/XXX.zip
    - amazon-s3-uploader@1.0.1:
        inputs:
        - file_path: "/Users/vagrant/app/XXX.zip"
        - aws_access_key: XXX
        - aws_secret_key: XXX
        - bucket_name: XXX
        - path_in_bucket: app
    - slack@2.3.0:
        inputs:
        - webhook_url: XXX
        - channel: "#builds"
        - from_username: XXX Build - Success
        - from_username_on_error: XXX Build - ERROR
        - message: |-
            Yay! New build is done! Here are results:

            Direct Download: <http://XXX.s3.amazonaws.com/app/XXX.zip>

            HockeyApp: <$HOCKEYAPP_DEPLOY_PUBLIC_URL>
        - message_on_error: |-
            Build Failed :(

            Bitrise.io: <$BITRISE_BUILD_URL>
    before_run: 
    after_run: 
app:
  envs:
  - opts:
      is_expand: false
    BITRISE_PROJECT_PATH: workspaces/XXX.xcworkspace
  - opts:
      is_expand: false
    BITRISE_SCHEME: Beta
```

I am running into following issues:
```
pupca:~/projects/XXX/bitrise $ bitrise --ci=true run Beta

  ██████╗ ██╗████████╗██████╗ ██╗███████╗███████╗
  ██╔══██╗██║╚══██╔══╝██╔══██╗██║██╔════╝██╔════╝
  ██████╔╝██║   ██║   ██████╔╝██║███████╗█████╗
  ██╔══██╗██║   ██║   ██╔══██╗██║╚════██║██╔══╝
  ██████╔╝██║   ██║   ██║  ██║██║███████║███████╗
  ╚═════╝ ╚═╝   ╚═╝   ╚═╝  ╚═╝╚═╝╚══════╝╚══════╝

Version: 1.4.2

INFO[00:54:59] bitrise runs in CI mode           
INFO[00:54:59] Running workflow: Beta            

INFO[00:54:59] Switching to workflow: Beta       

+------------------------------------------------------------------------------+
| (0) activate-ssh-key@3.1.1                                                   |
+------------------------------------------------------------------------------+
| id: activate-ssh-key                                                         |
| version: 3.1.1                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2016-10-19T00:54:59-04:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |

# Success

The SSH key was saved to */Users/pupca/.ssh/bitrise_step_activate_ssh_key*
and was successfully added to ssh-agent.
|                                                                              |
+----+--------------------------------------------------------------+----------+
| ✅  | activate-ssh-key@3.1.1                                       | 0.98 sec |
+----+--------------------------------------------------------------+----------+

                                          ▼

+------------------------------------------------------------------------------+
| (1) git-clone@3.4.0                                                          |
+------------------------------------------------------------------------------+
| id: git-clone                                                                |
| version: 3.4.0                                                               |
| collection: https://github.com/bitrise-io/bitrise-steplib.git                |
| toolkit: bash                                                                |
| time: 2016-10-19T00:55:00-04:00                                              |
+------------------------------------------------------------------------------+
|                                                                              |
/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go:15:2: cannot find package "github.com/bitrise-io/go-utils/cmdex" in any of:
	/usr/local/go/src/pkg/github.com/bitrise-io/go-utils/cmdex (from $GOROOT)
	/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/go-utils/cmdex (from $GOPATH)
/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go:16:2: cannot find package "github.com/bitrise-io/go-utils/errorutil" in any of:
	/usr/local/go/src/pkg/github.com/bitrise-io/go-utils/errorutil (from $GOROOT)
	/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/go-utils/errorutil (from $GOPATH)
/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go:17:2: cannot find package "github.com/bitrise-io/go-utils/pathutil" in any of:
	/usr/local/go/src/pkg/github.com/bitrise-io/go-utils/pathutil (from $GOROOT)
	/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/go-utils/pathutil (from $GOPATH)
|                                                                              |
+----+--------------------------------------------------------------+----------+
| 🚫  | git-clone@3.4.0 (exit code: 1)                               | 1.2 sec  |
+----+--------------------------------------------------------------+----------+
| Issue tracker: https://github.com/bitrise-io/steps-git-clone/issues          |
| Source: https://github.com/bitrise-io/steps-git-clone                        |
+----+--------------------------------------------------------------+----------+
```

The problem was resolved on slack channel and the cause was outdated version of GO (1.2.1). After update to GO 1.7.1 the problem was resolved. But I was asked to track this issue as CLI 1.4.2 should handle such issues itself

**TL;DR**
```
/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go:15:2: cannot find package "github.com/bitrise-io/go-utils/cmdex" in any of:
	/usr/local/go/src/pkg/github.com/bitrise-io/go-utils/cmdex (from $GOROOT)
	/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/go-utils/cmdex (from $GOPATH)
/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go:16:2: cannot find package "github.com/bitrise-io/go-utils/errorutil" in any of:
	/usr/local/go/src/pkg/github.com/bitrise-io/go-utils/errorutil (from $GOROOT)
	/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/go-utils/errorutil (from $GOPATH)
/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/git-clone/gitutil/gitutil.go:17:2: cannot find package "github.com/bitrise-io/go-utils/pathutil" in any of:
	/usr/local/go/src/pkg/github.com/bitrise-io/go-utils/pathutil (from $GOROOT)
	/var/folders/nq/mx509bjd37546m565w0t8snr0000gn/T/bitrise777601539/step_src/go/src/github.com/bitrise-io/go-utils/pathutil (from $GOPATH)
```

While on CLI 1.4.2, with old (1.2.1) version of GO. Resolved by updating GO to 1.7.1 but should be fixed anyway.

**Comment by viktorbenei** [comment]
Thanks a lot, we'll check what could cause this ASAP!

**Comment by viktorbenei** [comment]
Most likely this was because the Git Clone step does not (yet) utilize the CLI's Go toolkit - https://github.com/bitrise-io/steps-git-clone/blob/master/step.yml

So, it instead uses the system default Go (if any) during the step run

**Comment by viktorbenei** [comment]
I'll close this here, and created a separate ticket at https://github.com/bitrise-io/steps-git-clone/issues/25

Again, thanks for reporting! ;)
